src/main/java/com/github/manolo8/darkbot/Bot.java

```diff
@@ -32,10 +38,43 @@ public static void main(String[] args) throws IOException {
         LibSetup.setupLibraries();
         StartupParams params = new StartupParams(args);
 
+        checkUniqueInstance(params); //java 9
         checkJavaVersion(params);
         SwingUtilities.invokeLater(() -> new Main(params));
     }
 
+    private static void checkUniqueInstance(StartupParams params) throws IOException {
+        if (params.has(StartupParams.LaunchArg.NO_WARN)) return;
+
+        Path filePath = Paths.get("curr.pid");
+        long currentPid = ProcessHandle.current().pid();
+
+        if (Files.exists(filePath)) {
+            List<String> fileContent = new ArrayList<>(Files.readAllLines(filePath));
+
+            if (fileContent.isEmpty()) {
+                Files.writeString(filePath, "\n" + currentPid, StandardOpenOption.APPEND);
+            } else {
```

Comment:
```suggestion
            if (!fileContent.isEmpty()) {
```
Just go straight to checking